strawberryfields/api/connection.py

```diff
@@ -276,12 +276,31 @@ def get_job_result(self, job_id: str) -> Result:
                 buf.seek(0)
 
                 samples = np.load(buf, allow_pickle=False)
+                if isinstance(samples, np.lib.npyio.NpzFile):
+                    samples = dict(samples)  # convert to dict to free buffer
 
+            # Samples represent photon numbers.
+            # Convert to int64, to avoid unexpected behaviour
+            # when users postprocess these samples.
+            def int_to_int64(samples):
                 if np.issubdtype(samples.dtype, np.integer):
-                    # Samples represent photon numbers.
-                    # Convert to int64, to avoid unexpected behaviour
-                    # when users postprocess these samples.
-                    samples = samples.astype(np.int64)
+                    return samples.astype(np.int64)
+                return samples
+
+            if isinstance(samples, dict):
+                samples = {key: int_to_int64(value) for key, value in samples.items()}
```

Comment:
What would these keys be if returned from a QKD device?

Reply:
Haven't yet decided exactly what to call them, but we'd like to return two sets of numbers to users---the measured "results" (floats), and the implemented values that will likely differ from the requested values due to scaling and hardware limitations (complex floats).